californium-core/src/main/java/org/eclipse/californium/core/observe/InMemoryObservationStore.java

```diff
@@ -156,8 +162,8 @@ public synchronized void start() {
 		int healthStatusInterval = config.getInt(NetworkConfig.Keys.HEALTH_STATUS_INTERVAL,
 				NetworkConfigDefaults.DEFAULT_HEALTH_STATUS_INTERVAL); // seconds
 
-		if (healthStatusInterval > 0 && HEALTH_LOGGER.isDebugEnabled()) {
-			statusLogger = ExecutorsUtil.getScheduledExecutor().scheduleAtFixedRate(new Runnable() {
+		if (healthStatusInterval > 0 && HEALTH_LOGGER.isDebugEnabled() && executor != null) {
```

Comment:
So, if the secondary gets `null` (how ever), health status gets disabled.

Reply:
yep